src/classes/PS_IntegrationService_TEST.cls

```diff
@@ -35,16 +35,112 @@
+    private static Map<String, PS_IntegrationService.Config> paymentConfigParams;
+    private static final String API_KEY = 'apiKey';
+    private static final String PAYMENTS_APPLICATION_TYPE = 'payments';
+    private static final String MAKANA_APPLICATION_TYPE = 'makana';
+
+    static {
+        buildConfigurationParams(); // Build Test Configuration Parameters
+    }
+
     /*******************************************************************************************************
-     * @description tests storing of the secrets we receive from the Payment Services
-     * Integrations package
+     * @description Test for storing the secrets we receive from the Payment Services
+     * Integrations package with a valid API Key
      */
-    @isTest
-    private static void setConfigurationTest() {
-        // TODO This method can be implemented after the setConfiguration method is implemented
-         
+    @IsTest
+    private static void shouldSetConfigurationWithPaymentsAPIKey() {
+        Test.startTest();
+
+        PS_IntegrationService.setConfiguration(configParams);
+
+        List<Payment_Services_Configuration__c> configurations = [
+                SELECT Key__c, Service_Key__c, Value__c, Is_Secret__c, Service__c
+                FROM Payment_Services_Configuration__c
+        ];
+        Test.stopTest();
+        System.assert(!configurations.isEmpty());
+        for (Payment_Services_Configuration__c config : configurations) {
+         if (config.Service__c == PAYMENTS_APPLICATION_TYPE) {
+          System.assertEquals(paymentConfigParams.get(config.Key__c).value, config.Value__c);
+         }
+
+         if (config.Service__c == MAKANA_APPLICATION_TYPE) {
+          System.assertEquals(makanaConfigParams.get(config.Key__c).value, config.Value__c);
+         }
+
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Test for storing the secrets we receive from the Payment Services
+    * Integrations package with invalid configuration parameters
+    */
+    @IsTest
+    private static void shouldSetConfigurationWithInvalidConfigurationParameters() {
+       configParams = new Map<String, List<Map<String, String>>>();
+
+        Test.startTest();
+        try {
+            PS_IntegrationService.setConfiguration(configParams);
+        } catch (PS_IntegrationService.IntegrationServiceException ex) {
+            System.assertEquals(PS_IntegrationService.INVALID_CONFIG_PARAMS, ex.getMessage());
+        }
+        Test.stopTest();
+    }
+
+    /*******************************************************************************************************
+   * @description Helper method to build test configuration parameters
+   */
+    private static void buildConfigurationParams() {
+     List<Map<String, String>> applicationConfig = new List<Map<String, String>>();
```

Comment:
More mis-aligned indents - should be four spaces for this whole block from the first level of the method body